client/handlers.go

```diff
@@ -133,9 +133,10 @@ func (v2 *signer) Sign() error {
 	v2.Request.Header["x-amz-date"] = []string{v2.Time.In(time.UTC).Format(time.RFC1123)}
 
 	// Alibaba Cloud OSS date's formate must be http.TimeFormat
-	// URL Host's format is host or host:port
-	if config.Provider(strings.Split(host, ":")[0]) == "alicloud" {
+	// Alibaba Cloud OSS uses virtual hosted and the URL Host's format is <bucket-name>.host or <bucket-name>host:port
+	if config.Provider(strings.Join(strings.Split(strings.Split(host, ":")[0], ".")[1:], ".")) == "alicloud" {
```

Comment:
Hi, there are currently no `handlers_test.go` but could you add a new test file for this? Are `bucket.host` and `bucket.host:port` the only supported things? Would it be possible for something like `host` or `host:port` https://play.golang.org/p/cUo1Tqp7DmE

Also, the comment has a typo, it says `<bucket-name>host:port` instead of `<bucket-name>.host:port`.

Reply:
For virtual host style, only `bucket.host` and `bucket.host:port` are supported.